app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,22 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @user = current_user
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @user = current_user
+    @comment = Comment.create(params.require(:comment).permit(:text))
```

Comment:
**[OPTIONAL]**

- [ ]  It is always good practice to put `params.require(:comment).permit(:text)` in a method and make the method private. This is a security feature in rails. To learn more about this kindly refer to this lesson